pkg/scanner/utils/utils_test.go

```diff
@@ -0,0 +1,45 @@
+package utils
+
+import (
+	"testing"
+
+	"github.com/knqyf263/go-version"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMatchVersions2(t *testing.T) {
+	testCases := []struct {
+		name           string
+		currentVersion string
+		rangeVersion   []string
+		expectedCheck  bool
+	}{
+		{
+			name:           "pass: expect true when os/machine is in pre-release",
+			currentVersion: "1.9.25-x86-mingw32",
+			rangeVersion:   []string{`>= 1.9.24`},
```

Comment:
@knqyf263 I used https://github.com/Masterminds/semver, It passed all constraints reported in issue, also it will work with all possible node.js special constraints, please take a look at tests now.

Reply:
@knqyf263 please check

Reply:
Thanks!